e2e/framework/pages/details/details-page.ts

```diff
@@ -0,0 +1,40 @@
+import { Page } from "playwright";
+import { PageWithNavMenuAndMessages } from "../_common-page/common-page";
+import { waits } from "../../../libs/helpers/waits";
+
+export class DetailsPage extends PageWithNavMenuAndMessages {
+    private heroNameInputFieldSelector = '#hero-name';
+
+    constructor(page: Page) {
+        super(page);
+    }
+
+    async getHeroId(): Promise<string | undefined> {
+        let allTexts = await (await waits(this.page).waitVisibility('//div[child::span]')).textContent();
+        return allTexts?.split(":")[1].trim();
```

Comment:
If `allTexts` is undefined `.trim()` will fail, probably should also be `?.trim()`.

Reply:
Fixed.